investors/views.py

```diff
@@ -202,3 +211,64 @@ def perform_destroy(self, instance):
             extra={"saved_id": instance.pk, "by_user": self.request.user.pk},
         )
         super().perform_destroy(instance)
+
+class ViewedStartupPagination(pagination.PageNumberPagination):
+    """
+    Pagination class for recently viewed startups.
+    Default page size is 10, can be overridden via ?limit query parameter.
+    """
+    page_size = 10
+    page_size_query_param = "limit"
+    max_page_size = 50
+
+
+class ViewedStartupListView(generics.ListAPIView):
+    """
+    GET /api/v1/startups/viewed/
+    Retrieve a paginated list of recently viewed startups for the authenticated investor.
+    """
+    serializer_class = ViewedStartupSerializer
+    permission_classes = [IsAuthenticated, IsInvestor]
+    pagination_class = ViewedStartupPagination
+
+    def get_queryset(self):
+        return ViewedStartup.objects.filter(investor__user=self.request.user).order_by("-viewed_at")
+
+
+class ViewedStartupCreateView(APIView):
+    """
+    POST /api/v1/startups/view/{startup_id}/
+    Log that the authenticated investor has viewed a specific startup.
+    Return the serialized ViewedStartup instance.
+    """
+    permission_classes = [IsAuthenticated, IsInvestor]
+
+    def post(self, request, startup_id):
+        startup = get_object_or_404(Startup, id=startup_id)
+        investor = request.user.investor 
+
+        viewed_obj, created = ViewedStartup.objects.update_or_create(
+            investor=investor,
+            startup=startup,
+            defaults={"viewed_at": timezone.now()}
+        )
+
+        serializer = ViewedStartupSerializer(viewed_obj)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class ViewedStartupClearView(APIView):
+    """
+    DELETE /api/v1/startups/viewed/clear/
+    Clear the authenticated investor's viewed startups history.
+    Return number of deleted entries.
+    """
+    permission_classes = [IsAuthenticated, IsInvestor]
+
+    def delete(self, request):
+        investor = request.user.investor
+        deleted_count, _ = ViewedStartup.objects.filter(investor=investor).delete()
```

Comment:
Bulk delete does not clear related M2M relationships on Investor.viewed_startups. Consider clearing the M2M field explicitly to maintain consistency.
```
investor.viewed_startups.clear()
```